core/src/main/scala/org/broadinstitute/dsde/rawls/workspace/WorkspaceService.scala

```diff
@@ -365,12 +365,29 @@ class WorkspaceService(protected val userInfo: UserInfo,
     }
   }
 
+  private def deleteWorkspaceTransaction(workspaceName: WorkspaceName, workspaceContext: Workspace) = {
+    dataSource.inTransaction { dataAccess =>
+      for {
+        // Delete components of the workspace
+        _ <- dataAccess.submissionQuery.deleteFromDb(workspaceContext.workspaceIdAsUUID)
+        _ <- dataAccess.methodConfigurationQuery.deleteFromDb(workspaceContext.workspaceIdAsUUID)
+        _ <- dataAccess.entityQuery.deleteFromDb(workspaceContext.workspaceIdAsUUID)
+
+        // Delete the workspace
+        _ <- dataAccess.workspaceQuery.delete(workspaceName)
+
+        // Schedule bucket for deletion
+        _ <- dataAccess.pendingBucketDeletionQuery.save(PendingBucketDeletionRecord(workspaceContext.bucketName))
+      } yield ()
+    }
+  }
+
   private def deleteWorkspace(workspaceName: WorkspaceName, workspaceContext: Workspace): Future[PerRequestMessage] = {
     //Attempt to abort any running workflows so they don't write any more to the bucket.
     //Notice that we're kicking off Futures to do the aborts concurrently, but we never collect their results!
     //This is because there's nothing we can do if Cromwell fails, so we might as well move on and let the
     //ExecutionContext run the futures whenever
-    val deletionFuture: Future[Seq[WorkflowRecord]] =
+    val abortWorkflowFuture: Future[Seq[WorkflowRecord]] =
```

Comment:
Hooray for descriptive naming!  After reading it further though, might I suggest something like:  

```suggestion
    val gatherWorkflowsToAbort: Future[Seq[WorkflowRecord]] =
```

Optional:  I know there seems to be a convention in Rawls to use nested functions.  I personally feel like they complicate things.  I find shorter, well named functions easier to read/understand/follow.  So I might suggest extracting this out into its own private method, but totally up to you.  (for example, with the indenting and all the curly braces, I had a hard time telling in Github where the function ends)